TEST CASES FOR 7-base_geometry MODULE
=====================================

USAGE
=====

Importing the module
>>> BaseGeometry = __import__('7-base_geometry').BaseGeometry


Test Case 1: An integer value is entered
>>> bg = BaseGeometry()
>>> bg.integer_validator("my_int", 12)
>>> bg.integer_validator("width", 89)

Test Case 2: A string is used as value
>>> bg = BaseGeometry()
>>> bg.integer_validator("my_int", "Made")
Traceback (most recent call last):
	...
TypeError: my_int must be an integer


Test Case 3: Value == 0
>>> bg = BaseGeometry()
>>> bg.integer_validator("my_int", 0)
Traceback (most recent call last):
        ...
ValueError: my_int must be greater than 0

Test Case 3: Value less than 0
>>> bg = BaseGeometry()
>>> bg.integer_validator("my_int", -5)
Traceback (most recent call last):
        ...
ValueError: my_int must be greater than 0

Test Case 3: Value == none
>>> bg = BaseGeometry()
>>> bg.integer_validator("my_int", None)
Traceback (most recent call last):
        ...
TypeError: my_int must be an integer

Test Case 4: Value == boolean
>>> bg = BaseGeometry()
>>> bg.integer_validator("my_int", True)
Traceback (most recent call last):
        ...
TypeError: my_int must be an integer


Test Case 5: Value is a list
>>> bg = BaseGeometry()
>>> bg.integer_validator("my_int", [1, 5, 7])
Traceback (most recent call last):
        ...
TypeError: my_int must be an integer


Test Case 6: Value is a list
>>> bg = BaseGeometry()
>>> bg.integer_validator("my_int", (2, 8))
Traceback (most recent call last):
        ...
TypeError: my_int must be an integer

Test Case 7: Value == none
>>> bg = BaseGeometry()
>>> bg.integer_validator("my_int", None)
Traceback (most recent call last):
        ...
TypeError: my_int must be an integer

Test Case 8: Value is a dictionary
>>> bg = BaseGeometry()
>>> bg.integer_validator("my_int", {"val": 20})
Traceback (most recent call last):
        ...
TypeError: my_int must be an integer

Test Case 9: One argument is provided
>>> bg = BaseGeometry()
>>> bg.integer_validator("my_int")
Traceback (most recent call last):
        ...
TypeError: integer_validator() missing 1 required positional argument: 'value'

Test Case 9: No argument is provided
>>> bg = BaseGeometry()
>>> bg.integer_validator()
Traceback (most recent call last):
        ...
TypeError: integer_validator() missing 2 required positional arguments: 'name' and 'value'
